checkov/terraform/checks/resource/aws/DLMScheduleCrossRegionEncryption.py

```diff
@@ -17,12 +17,13 @@ def scan_resource_conf(self, conf):
             if policy.get("schedule") and isinstance(policy.get("schedule"), list):
                 schedules = policy.get("schedule")
                 for idx, schedule in enumerate(schedules):
-                    if schedule.get("cross_region_copy_rule") and isinstance(schedule.get("cross_region_copy_rule"), list):
-                        cross = schedule.get("cross_region_copy_rule")[0]
-                        if cross.get("encrypted") == [True]:
-                            return CheckResult.PASSED
-                        self.evaluated_keys = [f"policy_details/schedule/{idx}/cross_region_copy_rule/encrypted"]
-                        return CheckResult.FAILED
+                    if schedule.get("cross_region_copy_rule") and isinstance(schedule.get("cross_region_copy_rule"), list) and all(isinstance(rule, dict) for rule in schedule.get("cross_region_copy_rule")):
+                        for c_idx, cross_schedule_rule in enumerate(schedule.get("cross_region_copy_rule")):
+                            if cross_schedule_rule.get("encrypted") != [True]:
```

Comment:
```suggestion
                    if schedule.get("cross_region_copy_rule") and isinstance(schedule.get("cross_region_copy_rule"), list):
                        for c_idx, cross_schedule_rule in enumerate(schedule.get("cross_region_copy_rule")):
                            if isinstance(rule, dict) and cross_schedule_rule.get("encrypted") != [True]:
```
how about doing it like this?

Reply:
@gruebel I prefer checking it out of the for loop, so the check result would be UNKNOWN